### PR TITLE
Add dotty runner script able to run dotty-compiled class with `main`

### PR DESCRIPTION
**Pull request by felixmulder**
Addresses #1208

Review: @DarkDimius / @smarter

### CODE REVIEW

File: bin/dotty [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+DOTTY_ROOT="`( cd \"$DOTTY_ROOT\" && pwd )`/.."  # absolute
+
+# CLASS_PATH is derived from the DOTTY_ROOT
+CLASS_PATH="-J-Xbootclasspath/a:.:$DOTTY_ROOT/target/scala-2.11/classes/"
+
+function runMain {
+    scala $CLASS_PATH $@
+}
+
+if [ -z "$1" ]; then
```
Comment (smarter) [inline bin/dotty]: I think no arguments should launch the REPL.

Comment (felixmulder) [inline bin/dotty]: Good idea! Will update the PR

File: bin/dotty [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# Try to autodetect real location of the script
+DOTTY_ROOT="`readlink \"$0\"`"             #  relative, symbolic links resolved
```
Comment (smarter) [inline bin/dotty]: `DOTTY_ROOT="$(readlink "$0")"` (no need to escape `"` inside `$(...)`)

Comment (felixmulder) [inline bin/dotty]: Changed it in the ´dotc´ script as well

**Comment by DarkDimius** [comment]
I'd better not call this script `dotty`. `dotty` is part of graphviz library that is very common in unix environments.

**Comment by felixmulder** [comment]
@DarkDimius, sure - if it's put on `$PATH` that'd be bad. Suggestions on the name? `dotr`?

**Comment by DarkDimius** [comment]
I'm using the script `dottyr` on my machine.

**Comment by felixmulder** [comment]
As discussed in the meeting, change the name of the script to `dotr`

**Comment by DarkDimius** [comment]
needs rebasing, and a patch to silence `cd` in `dotr`.
Otherwise LGTM.

**Pull request merged**